components/widgets/menu/menu.jsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {isMobile} from 'utils/utils';
+
+export default class Menu extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.node,
+        openLeft: PropTypes.bool,
+        openUp: PropTypes.bool,
+        id: PropTypes.string,
+    };
+
+    constructor(props) {
+        super(props);
+        this.node = React.createRef();
+    }
+
+    // Used from DotMenu component to know in which direction show the menu
+    rect() {
+        if (this.node && this.node.current) {
+            return this.node.current.getBoundingClientRect();
+        }
+        return null;
+    }
+
+    render() {
+        const {children, openUp, openLeft, id} = this.props;
+        const styles = {};
+        if (openLeft && !isMobile()) {
+            styles.left = 'inherit';
+            styles.right = 0;
+        }
+        if (openUp && !isMobile()) {
+            styles.bottom = '100%';
+            styles.top = 'auto';
+        }
+
+        return (
+            <ul
+                id={id}
+                ref={this.node}
+                className='Menu dropdown-menu'
+                style={styles}
+                role='menu'
+                aria-labelledby='channel_header_dropdown'
```

Comment:
Why `channel_header_dropdown` here?

Reply:
that comes from refactoring the original probably makes sense to remove the aria label or add it as a property.

Reply:
Yeah, I think making it a prop makes sense (seems like something we wouldn't want to use as it's for screen reader accessibility).

Reply:
Yes, actually I have change it to use `aria-label` instead of `aria-labelledby` because there wasn't the id pointed there and actually, the menus itself haven't a proper element to label them, so I used `aria-label` and made it required to be sure all the menus in the app are accesible, as least by screen readers.